types/mongodb/index.d.ts

```diff
@@ -3869,36 +3917,73 @@ export type CollectionMapFunction<TSchema> = (this: TSchema) => void;
 
 export type CollectionReduceFunction<TKey, TValue> = (key: TKey, values: TValue[]) => TValue;
 
-/** @see https://mongodb.github.io/node-mongodb-native/3.6/api/Collection.html#~WriteOpResult */
+/**
+ * Returning object from write operations
+ *
+ * @see https://mongodb.github.io/node-mongodb-native/3.6/api/Collection.html#~WriteOpResult
+ */
 export interface WriteOpResult {
     ops: any[];
     connection: any;
     result: any;
 }
 
-/** @see https://mongodb.github.io/node-mongodb-native/3.6/api/Cursor.html#~resultCallback */
+/**
+ * Callback for cursor operations
+ *
+ * @see https://mongodb.github.io/node-mongodb-native/3.6/api/Cursor.html#~resultCallback
+ */
 export type CursorResult = object | null | boolean;
 
 type Default = any;
 type DefaultSchema = any;
 
-/** @see https://mongodb.github.io/node-mongodb-native/3.6/api/Cursor.html */
+/**
+ * Creates a new Cursor instance (INTERNAL TYPE, do not instantiate directly)
+ *
+ * @see https://mongodb.github.io/node-mongodb-native/3.6/api/Cursor.html
+ */
 export class Cursor<T = Default> extends Readable {
     [Symbol.asyncIterator](): AsyncIterableIterator<T>;
     sortValue: string;
     timeout: boolean;
     readPreference: ReadPreference;
-    /** @see https://mongodb.github.io/node-mongodb-native/3.6/api/Cursor.html#addCursorFlag */
+    /**
+     * Add a cursor flag to the cursor
+     *
+     * @param flag The flag to set, must be one of following ['`tailable`', '`oplogReplay`', '`noCursorTimeout`', '`awaitData`', '`partial`'].
+     * @param value The flag boolean value.
+     * @see https://mongodb.github.io/node-mongodb-native/3.6/api/Cursor.html#addCursorFlag
+     */
     addCursorFlag(flag: string, value: boolean): Cursor<T>;
-    /** @see https://mongodb.github.io/node-mongodb-native/3.6/api/Cursor.html#addQueryModifier */
-    addQueryModifier(name: string, value: boolean): Cursor<T>;
-    /** @see https://mongodb.github.io/node-mongodb-native/3.6/api/Cursor.html#batchSize */
+    /**
+     * Add a query modifier to the cursor query
+     *
+     * @param name The query modifier (must start with $, such as `$orderby` etc)
+     * @param value The modifier value.
+     * @see https://mongodb.github.io/node-mongodb-native/3.6/api/Cursor.html#addQueryModifier
+     */
+    addQueryModifier(name: string, value: boolean|string|number): Cursor<T>;
```

Comment:
Seems to me that changing the type of value to boolean | string | number caused the issue that is making the CI fail. Unfortunately, I cannot undo that as those are the correct types found on the documentation as of v3.6 so I believe mongoose will have to fix that error themselves. Will tag the author of the latest mongoose PR @serhalp so that he can take a look at this.

Reply:
I believe you can leave `value: boolean` and make a separate PR fixing the type.

Reply:
But the broken tests are in mongoose and not mongodb, therefore someone else has to fix those because it’s usually not good practice to change multiple packages in one